test/signalrclient-e2e-tests/hub_connection_tests.cpp

```diff
@@ -180,4 +180,57 @@ TEST(hub_connection_tests, send_message_after_reconnect)
     ASSERT_FALSE(received_event->wait(2000));
 
     ASSERT_EQ(*message, U("[\"Send: test\"]"));
-}
\ No newline at end of file
+}
+
+TEST(hub_connection_tests, progress_report)
+{
+    auto hub_conn = std::make_shared<signalr::hub_connection>(url);
+    std::vector<int> vec;
+
+    auto hub_proxy = hub_conn->create_hub_proxy(U("hubConnection"));
+
+    hub_conn->start().then([&hub_proxy, &vec]()
+    {
+        return hub_proxy.invoke<web::json::value>(U("invokeWithProgress"), [&vec](const web::json::value& arguments)
+        {
+            vec.push_back(arguments.as_integer());
+        });
+
+    }).get();
+
+    ASSERT_EQ(vec.size(), 10);
+
+    for (int i = 0; i < (int)vec.size(); i++)
```

Comment:
do `for (size_t i = 0; i < vec.size(); i++)` (or maybe even `for (auto i = 0; i < vec.size(); i++)` - but I don't think it would work - i.e. you would still get the warning)